nixos/lib/test-driver/default.nix

```diff
@@ -25,13 +27,19 @@ python3Packages.buildPythonApplication {
     setuptools
   ];
 
+  postPatch = ''
+    substituteInPlace test_driver/debug.py \
+      --replace-fail "@attach@" "${breakpointHook.attach}"
```

Comment:
instead of text-patching this, we could make the test driver parameter accept some kind of `--debug-hook ${breakpointHook.attach}` string and then use that in the `Debug` constructor. Not sure if overkill but not doing text patching might be nice and then it's also clear that the whole debug object is useless without this parameter.

Reply:
For me, the current approach was sufficient because I only iterated on this by `nix-build`ing the package (and was kinda "packager-brained" by patching sources when store-path references are needed).

So, while your solution is definitely cleaner, I don't see a strong reason to do so.

Hence, I'm fine with either option.